sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java

```diff
@@ -0,0 +1,99 @@
+package org.sormas.e2etests.steps.web.application.immunizations;
+
+import static org.sormas.e2etests.pages.application.cases.EditCasePage.*;
+import static org.sormas.e2etests.pages.application.immunizations.CreateNewImmunizationPage.MEANS_OF_IMMUNIZATIONS_COMBOBOX;
+import static org.sormas.e2etests.pages.application.immunizations.EditImmunizationPage.DISEASE_COMBOBOX;
+import static org.sormas.e2etests.pages.application.immunizations.EditImmunizationPage.FACILITY_COMBOBOX_IMMUNIZATION_INPUT;
+import static org.sormas.e2etests.pages.application.immunizations.EditImmunizationPage.FACILITY_NAME_DESCRIPTION_VALUE;
+import static org.sormas.e2etests.pages.application.users.EditUserPage.*;
+
+import cucumber.api.java8.En;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import javax.inject.Inject;
+import javax.inject.Named;
+import lombok.SneakyThrows;
+import org.assertj.core.api.SoftAssertions;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.pojo.web.Immunization;
+import org.sormas.e2etests.services.ImmunizationService;
+
+public class EditImmunizationSteps implements En {
+
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("M/d/yyyy");
+    public static Immunization aImmunization;
+    private final WebDriverHelpers webDriverHelpers;
+    private final SoftAssertions softly;
+
+    @SneakyThrows
+    @Inject
+    public EditImmunizationSteps(
+            WebDriverHelpers webDriverHelpers,
+            ImmunizationService immunizationService,
+            SoftAssertions softly,
+            @Named("ENVIRONMENT_URL") String environmentUrl) {
+        this.webDriverHelpers = webDriverHelpers;
+        this.softly = softly;
+
+        When(
+                "I check the created data is correctly displayed on Edit immunization page",
+                () -> {
+                    aImmunization = collectImmunizationData();
+
+                    softly
```

Comment:
I know that in a lot of places, the message lacks, so sometimes in allure we can't see exactly the reason of the failure, just the different value. Please add also the message for assert. I have a task to update all soft asserts from framework and add the messages